lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -86,6 +86,12 @@ bool swift::isExported(const ValueDecl *VD) {
 }
 
 static bool hasConformancesToPublicProtocols(const ExtensionDecl *ED) {
+  // Extensions of protocols cannot introduce additional conformances.
```

Comment:
If the extended nominal is null, the extension is invalid so you can skip everything.